localgov_alert_banner.module

```diff
@@ -75,3 +76,48 @@ function localgov_alert_banner_theme_suggestions_localgov_alert_banner(array $va
   $suggestions[] = 'localgov_alert_banner__' . $entity->id() . '__' . $sanitized_view_mode;
   return $suggestions;
 }
+
+/**
+ * Implements hook_modules_installed().
+ */
+function localgov_alert_banner_modules_installed($modules, $is_syncing) {
+
+  // Configure scheduled transitions if it's being installed.
+  if (in_array('scheduled_transitions', $modules)) {
+    localgov_alert_banner_configure_scheduled_transitions();
+  }
+}
+
+/**
+ * Configure scheduled transitions.
+ */
+function localgov_alert_banner_configure_scheduled_transitions() {
+
+  // Configure scheduled transitions for alert banners.
+  $scheduled_transitions_config = \Drupal::service('config.factory')->getEditable('scheduled_transitions.settings');
+  $alert_banner_bundles = \Drupal::service('entity_type.bundle.info')->getBundleInfo('localgov_alert_banner');
+  $bundles = [];
+  foreach ($alert_banner_bundles as $bundle => $info) {
+    $bundles[] = [
+      'entity_type' => 'localgov_alert_banner',
+      'bundle' => $bundle,
+    ];
+  }
+  $scheduled_transitions_config->set('bundles', $bundles);
+  $scheduled_transitions_config->save();
+  Cache::invalidateTags([
+    'scheduled_transition_settings',
+    'config:scheduled_transitions.settings',
+  ]);
+
+  // Add scheduling permissions to the emergency publisher user.
+  $permissions = [];
+  foreach ($alert_banner_bundles as $bundle => $info) {
+    $permissions = array_merge($permissions, [
+      'add scheduled transitions localgov_alert_banner ' . $bundle,
+      'reschedule scheduled transitions localgov_alert_banner ' . $bundle,
+      'view scheduled transitions localgov_alert_banner ' . $bundle,
+    ]);
+  }
+  user_role_grant_permissions('emergency_publisher', $permissions);
+}
```

Comment:
This code configures the Scheduled Transitions module for alert banners and adds the permissions to the alert banner provided emergency publisher user. It's not related to LocalGov Workflows and would add an unnecessary dependency between this module and workflows. What would be gained by moving this code?

Reply:
This is the pattern we've used elsewhere for installing additional functionality based on optional module dependencies: Both a check for the supporting module on hook_install of the module, and hook_modules_installed() when other modules are installed. We've then run a function and/or imported optional yml config.

Reply:
@ekes am I right in thinking that you mean we could support this in localgov_workflows by checking if alert banner is installed. I seem to remember that is a pattern used in directories (or events) to support some extra features.

For now this is ok as its a hook, so its more a curosity. However I am a little concerned about increasing the complexity of this module when it does have a mission critical function of being able to quickly put an emergency alert at the top of the page. So it was to ask how plausable could be moving the implementation entierley to lcoalgov_workflows so that handles everything to do with scheduling / workflows and if you dont install it nothing changes.

Reply:
If we're using Scheduled Transitions to do the scheduling of alert banners then it needs configuring, that is what this code does. It doesn't really have any relevance to Workflows (other than that also uses Scheduled Transitions). This code also only runs when the Scheduled Transitions module is installed, so don't really see how this increases complexity.